docs/guide/custom-functions.md

```diff
@@ -46,25 +71,89 @@ import { FunctionPlugin } from 'hyperformula';
+
+### Argument validation options
+
+In an optional `parameters` object, you can set rules for your function's argument validation.
+
+```javascript
+// import FunctionPlugin
+import { FunctionPlugin } from 'hyperformula';
+
+// start creating a class
+class CountHF extends FunctionPlugin {
+
+// define functions inside this plugin
+  public static implementedFunctions = {
+    'HYPER': {
+      // this method's functionality will be defined below
+      method: 'hyper',
+      // set your argument validation options below
+      parameters: {
+        passSubtype: false,
+        defaultValue: 10,
+        optionalArg: false,
+        minValue: 5,
+        maxValue: 15,
+        lessThan: 15,
+        greaterThan: 5
+      },
     }
   };
 }
 ```
 
-Similarly, there are other useful properties. 
-`isDependentOnSheetStructureChange` marks functions that need to be recalculated with 
-each change of the shape of the engine sheets. 
-`doesNotNeedArgumentsToBeComputed` marks functions that treat references or ranges in their arguments
-as arguments that do not create dependency. Other arguments are properly evaluated.
-`arrayFunction` denotes functions that enable array arithmetic in its arguments and nested expressions.
-`vectorizationForbidden` when set prevents function from ever being vectorized (however, it is up to implementation of a function to properly handle vectorization).
+You can set the following argument validation options:
+
+| Parameter | Type | Description |
+| --- | --- | --- |
+| `passSubtype` | Boolean | If set to `true`, arguments are passed with full type information.<br>(e.g. for numbers: `Date` or `DateTime` or `Time` or `Currency` or `Percentage`) |
+| `defaultValue` | `InternalScalarValue` \| `RawScalarValue` | If set to any value: if an argument is missing, its value defaults to `defaultValue`. |
+| `optionalArg` | Boolean | If set to `true`: if an argument is missing, and no `defaultValue` is set, the argument defaults to `undefined` (instead of throwing an error).<br><br>Setting this option to `true` is the same as setting `defaultValue` to `undefined`. |
+| `minValue` | Number | If set, numerical arguments need to be greater than or equal to `minValue`. |
+| `maxValue` | Number | If set, numerical arguments need to be less than or equal to `maxValue`. |
+| `lessThan` | Number | If set, numerical argument need to be less than `lessThan`. |
+| `greaterThan` | Number | If set, numerical argument need to be greater than `greaterThan`. |
+
+#### Handling missing arguments
+
+The `defaultValue` and `optionalArg` options let you decide what happens when a user doesn't pass enough valid arguments to your custom function.
+
+When you set `defaultValue` to any value, the `optionalArg` setting doesn't matter.
```

Comment:
Should we say it explicitly that `defaultValue` always makes it an optional arg?

Reply:
@wojciechczerniak - done: 3776f72